utils/src/activityLog/activities/ExecuteActivity.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { TreeItemCollapsibleState } from "vscode";
+import * as types from '../../../index';
+import { AppResource } from "../../../unified";
+import { localize } from "../../localize";
+import { AzExtParentTreeItem } from "../../tree/AzExtParentTreeItem";
+import { GenericTreeItem } from "../../tree/GenericTreeItem";
+import { nonNullProp } from "../../utils/nonNull";
+import { ActivityBase } from "../Activity";
+
+interface ExecuteActivityData<C extends types.ExecuteActivityContext> {
+    context: C;
+}
+
+export class ExecuteActivity<C extends types.ExecuteActivityContext> extends ActivityBase<void> {
+
+    public constructor(private readonly data: ExecuteActivityData<C>, task: types.ActivityTask<void>) {
+        super(task);
+    }
+
+    public initialState(): types.ActivityTreeItemOptions {
+        return {
+            label: this.label,
+            collapsibleState: TreeItemCollapsibleState.None,
+        }
+    }
+
+    public successState(): types.ActivityTreeItemOptions {
+        return {
+            label: this.label,
+            collapsibleState: this.data.context['activityResult'] ? TreeItemCollapsibleState.Expanded : TreeItemCollapsibleState.None,
+            getChildren: (parent: AzExtParentTreeItem) => {
+                if (this.data.context['activityResult']) {
```

Comment:
Since it's now a proper field in `context`:
```suggestion
                if (this.data.context.activityResult) {
```